### ISSUE TITLE
Kill background process jobs that timeout but ignore SIGTERM

### ISSUE THREAD

**Issue by jfrost-mo** [cylc/cylc-flow#5768]
### Problem

Currently a background job that intercepts SIGTERM does not get killed when the time limit expires. This is due to the timeout command only sending a SIGTERM by default.

### Proposed Solution

The timeout command provides a `--kill-after=` option to deal with this, which send the non-interceptable SIGKILL to the process after a number of seconds beyond the usual time limit. In line with [Slurm](https://slurm.schedmd.com/slurm.conf.html#OPT_KillWait), 30 seconds is probably reasonable.

**Comment by dpmatthews** [comment]
This option is only available on recent versions of `timeout` (e.g. it works on RHEL7 but not RHEL6) so I think it would have to be configurable

**Comment by jfrost-mo** [comment]
Fair enough. RHEL 6 is currently in final extra extended support, which ends on June 30, 2024. So any check could be removed after then, as people will have had 15 years to upgrade.
